public/chat_header_button.tsx

```diff
@@ -65,7 +65,9 @@ export const HeaderChatButton: React.FC<HeaderChatButtonProps> = (props) => {
       selectedTabId,
       preSelectedTabId,
       setSelectedTabId: (tabId: TabId) => {
-        setPreSelectedTabId(selectedTabId);
+        if (selectedTabId !== tabId) {
```

Comment:
How about add this if check to `use_chat_actions`? We don't need to call setSelectedTabId if selectedTabId is already "trace".